sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/ServicePrincipalsImpl.java

```diff
@@ -61,9 +61,10 @@ public ServicePrincipal getByName(String spn) {
 
     @Override
     public Mono<ServicePrincipal> getByNameAsync(final String name) {
-        return listByFilterAsync(String.format("servicePrincipalNames/any(c:c eq '%s')", name))
+        return listByFilterAsync(String.format("displayName eq '%s'", name))
             .singleOrEmpty()
-            .switchIfEmpty(listByFilterAsync(String.format("displayName eq '%s'", name)).singleOrEmpty());
+            .switchIfEmpty(
+                listByFilterAsync(String.format("servicePrincipalNames/any(c:c eq '%s')", name)).singleOrEmpty());
```

Comment:
Make sure this still works: `servicePrincipalNames/any(c:c eq '%s')`

Reply:
It works.